### PR TITLE
feat: schema-builder and resolver-builder can handle relations

### PR DESCRIPTION
**Pull request by magrinj**
This pull request implements the new `relations` feature as introduced in PR #2388 and addresses the issues highlighted in ticket #2143.

The enhancement now permits the execution of complex queries, such as those involving a `one-to-many` relationship. Below is a sample GraphQL query that demonstrates this capability:

```graphql
query Documents($firstDocument: Int, $firstSupplier: Int) {
  documents(first: $firstDocument) {
    edges {
      node {
        id
        title
        suppliers(first: $firstSupplier) {
          edges {
            node {
              id
              name
            }
          }
        }
      }
    }
  }
}
```

Here is a screenshot for reference:
Screenshot of GraphQL query example

## Technical Details

- We have improved efficiency by reducing the need for multiple requests. This is achieved by inferring arguments directly from the `GraphQLContextInfo` of nested relations, streamlining the querying process.

### CODE REVIEW

File: server/src/tenant/query-builder/factories/fields-string.factory.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    const fieldMetadataMap = new Map(
+      fieldMetadataCollection.map((metadata) => [metadata.name, metadata]),
+    );
+
+    for (const [fieldKey, fieldValue] of Object.entries(selectedFields)) {
+      let fieldAlias: string | null;
+
+      if (fieldMetadataMap.has(fieldKey)) {
+        // We're sure that the field exists in the map after this if condition
+        // ES6 should tackle that more peroperly
```
Comment (Weiko) [inline server/src/tenant/query-builder/factories/fields-string.factory.ts:51]: typo

File: server/src/tenant/query-builder/__tests__/pg-graphql-query-builder.spec.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
+// import { GraphQLResolveInfo } from 'graphql';
+
+// import { FieldMetadataTargetColumnMap } from 'src/metadata/field-metadata/interfaces/field-metadata-target-column-map.interface';
```
Comment (Weiko) [inline server/src/tenant/query-builder/__tests__/pg-graphql-query-builder.spec.ts:3]: It is intended? If the whole file is commented let's not push it maybe?

**Review by Weiko — APPROVED** [review]
Awesome, LGTM 👏

File: server/src/metadata/relation-metadata/relation-metadata.entity.ts [hunk 3]
```diff
@@ -17,10 +17,12 @@ import {
   Relation,
 } from '@ptc-org/nestjs-query-graphql';
 
+import { RelationMetadataInterface } from 'src/tenant/schema-builder/interfaces/relation-metadata.interface';
+
 import { FieldMetadata } from 'src/metadata/field-metadata/field-metadata.entity';
 import { ObjectMetadata } from 'src/metadata/object-metadata/object-metadata.entity';
 
-export enum RelationType {
+export enum RelationMetadataType {
```
[the remark this answers is not in the document]
Comment (Weiko) [reply server/src/metadata/relation-metadata/relation-metadata.entity.ts:25]: I agree! 👍

**Comment by charlesBochet** [comment]
I'm getting errors: 
image

**Comment by charlesBochet** [comment]
Filter is mandatory:
image

**Comment by magrinj** [comment]
@charlesBochet Hmm, I'll take a look on that tomorrow, I didn't get time to test everything like standards objects

**Comment by charlesBochet** [comment]
Ok, no problem! I'm adding the bug I find :) It will wait tomorrow
image

**Comment by magrinj** [comment]
@charlesBochet And if you don't provide the filter argument ?

**Comment by charlesBochet** [comment]
Here you go :D 
image

**Pull request merged**